Milestone 2 updated

pasivinay opened the pull request:
### **This PR fixes issues in milestone 2:**

1. Modified Dockerfile for removing redundant layers
2. Modified .gitignore based on folder structure change
3. Added dependencies to makefile target
4. Modified README file to remove branch dependency while cloning

pasivinay committed:
Rename app/requirements.txt to requirements.txt

Repositioned requirements.txt file from app folder to base folder.

pasivinay committed:
Modify Dockefile to remove redundant layers

Modifies dockerfile to remove redundant COPY layers and utilizes image
from build layer.

pasivinay committed:
Modify makefile to add dependencies

Added dependencies to build and test targets and modified migrate target
to incorporate changes in application directory structure.

pasivinay committed:
Add .dockerignore file

Added .dockerignore file to exclude unnecessary files from being copied
to docker image.

pasivinay committed:
Move app/.example.env to base folder

Moves .example.env file from app to base folder.

pasivinay committed:
Modify .gitignore file

Modifies .gitignore file

pasivinay committed:
Modify README file

Modifies README.md file to remove --branch option from git clone command
in setup insstructions section.

rajeshecb70 commented on milestone_2/Dockerfile:
create a non-root user and run docker images.

pasivinay replied:
### **Following changes have been made:**

1.  Added non-root user for flask application in dockerfile
2. Added checks in makefile for db migrations
3. Added note to use semver tagging in README.md file

pasivinay committed:
Modify Dockerfile to add dedicated user for flask

Adds a RUN statement to add a dedicated user for flask application and
grant ownership of app folder. Also adds a USER keyword for setting
flask as the user to run commands

pasivinay committed:
Modify Makefile to add check for pending migration

Added checks for pending migration and already initialized migrations.

pasivinay committed:
Modify README.md to add note for semver tagging

Added a note for users to follow semver versioning for tagging docker
images.

rajeshecb70 reviewed (approved):
It looks good to me.
